assets/src/edit-story/components/inspector/document/test/status.js

```diff
@@ -0,0 +1,109 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { render, fireEvent } from '@testing-library/react';
+import { ThemeProvider } from 'styled-components';
+
+/**
+ * Internal dependencies
+ */
+import StoryContext from '../../../../app/story/context';
+import InspectorContext from '../../../inspector/context';
+import theme from '../../../../theme';
+import StatusPanel from '../status';
+
+function setupPanel(
+  capabilities = {
+    hasPublishAction: true,
+  }
+) {
+  const updateStory = jest.fn();
+  const deleteStory = jest.fn();
+  const loadStatuses = jest.fn();
+  const loadUsers = jest.fn();
+
+  const storyContextValue = {
+    state: {
+      story: { status: 'draft', password: '' },
+      capabilities,
+    },
+    actions: { updateStory, deleteStory },
+  };
+  const statuses = [
+    {
+      value: 'draft',
+      name: 'Draft',
+    },
+    {
+      value: 'publish',
+      name: 'Public',
+    },
+    {
+      value: 'private',
+      name: 'Private',
+    },
+  ];
+  const inspectorContextValue = {
+    actions: { loadStatuses, loadUsers },
+    state: { statuses },
+  };
+  const { getByText, queryByText } = render(
+    <ThemeProvider theme={theme}>
+      <StoryContext.Provider value={storyContextValue}>
+        <InspectorContext.Provider value={inspectorContextValue}>
+          <StatusPanel />
+        </InspectorContext.Provider>
+      </StoryContext.Provider>
+    </ThemeProvider>
+  );
+  return {
+    getByText,
+    queryByText,
+    updateStory,
+  };
+}
+
+describe('StatusPanel', () => {
+  it('should render Status Panel', () => {
+    const { getByText } = setupPanel();
+    const element = getByText('Status & Visibility');
+    expect(element).toBeDefined();
+
+    const radioOption = getByText('Draft');
+    expect(radioOption).toBeDefined();
+  });
+
+  it('should not render the status option without correct permissions', () => {
+    const { queryByText } = setupPanel({
+      hasPublishAction: false,
+    });
+    expect(queryByText('Draft')).toBeNull();
+  });
+
+  it('should update the story when clicking on status', () => {
+    const { getByText, updateStory } = setupPanel();
+    const publishOption = getByText(/Public/i).closest('label');
+    fireEvent.click(publishOption);
+    expect(updateStory).toHaveBeenCalledWith({
+      properties: {
+        status: 'publish',
+        password: '',
+      },
+    });
+  });
```

Comment:
Needs a test for deleting story too.

Reply:
The delete button is not actually in Figma and it's not clear yet where it should be but I guess it doesn't harm to add a test nevertheless :) Added in 2feebc3